packages/astro/src/core/build/plugins/plugin-css.ts

```diff
@@ -25,236 +25,312 @@ interface PluginOptions {
-				const createNameForParentPages = namingIncludesHash
-					? assetName.shortHashedName
-					: assetName.createSlugger(settings);
-
-				extendManualChunks(outputOptions, {
-					after(id, meta) {
-						// For CSS, create a hash of all of the pages that use it.
-						// This causes CSS to be built into shared chunks when used by multiple pages.
-						if (isBuildableCSSRequest(id)) {
-							for (const [pageInfo] of walkParentInfos(id, {
-								getModuleInfo: meta.getModuleInfo,
-							})) {
-								if (new URL(pageInfo.id, 'file://').searchParams.has(PROPAGATED_ASSET_FLAG)) {
-									// Split delayed assets to separate modules
-									// so they can be injected where needed
-									return createNameHash(id, [id]);
-								}
-							}
-							return createNameForParentPages(id, meta);
+			const assetFileNames = outputOptions.assetFileNames;
+			const namingIncludesHash = assetFileNames?.toString().includes('[hash]');
+			const createNameForParentPages = namingIncludesHash
+				? assetName.shortHashedName
+				: assetName.createSlugger(settings);
+
+			extendManualChunks(outputOptions, {
+				after(id, meta) {
+					// For CSS, create a hash of all of the pages that use it.
+					// This causes CSS to be built into shared chunks when used by multiple pages.
+					if (isBuildableCSSRequest(id) === false) return;
+					for (const [pageInfo] of walkParentInfos(id, {
+						getModuleInfo: meta.getModuleInfo,
+					})) {
+						if (new URL(pageInfo.id, 'file://').searchParams.has(PROPAGATED_ASSET_FLAG)) {
+							// Split delayed assets to separate modules
+							// so they can be injected where needed
+							return createNameHash(id, [id]);
 						}
-					},
-				});
-			},
+					}
+					return createNameForParentPages(id, meta);
+				},
+			});
+		},
 
-			async generateBundle(_outputOptions, bundle) {
-				type ViteMetadata = {
-					importedAssets: Set<string>;
-					importedCss: Set<string>;
-				};
+		async generateBundle(_outputOptions, bundle) {
+			for (const [_, chunk] of Object.entries(bundle)) {
+				if (chunk.type !== 'chunk') continue;
+				if ('viteMetadata' in chunk === false) continue;
+				const meta = chunk.viteMetadata as ViteMetadata;
 
-				const appendCSSToPage = (
-					pageData: PageBuildData,
-					meta: ViteMetadata,
-					depth: number,
-					order: number
-				) => {
-					for (const importedCssImport of meta.importedCss) {
-						// CSS is prioritized based on depth. Shared CSS has a higher depth due to being imported by multiple pages.
-						// Depth info is used when sorting the links on the page.
-						if (pageData?.css.has(importedCssImport)) {
-							// eslint-disable-next-line
-							const cssInfo = pageData?.css.get(importedCssImport)!;
-							if (depth < cssInfo.depth) {
-								cssInfo.depth = depth;
-							}
+				// Chunks that have the viteMetadata.importedCss are CSS chunks
+				if (meta.importedCss.size < 1) continue;
 
```

Comment:
nit: The comment here is also contradicting the condition check, but since this helps prevent a large indent, maybe we can update the comment:

```js
// Skip if the chunk has no CSS, we want to handle CSS chunks only
```